src/librustc_resolve/macros.rs

```diff
@@ -277,11 +275,12 @@ impl<'a, 'crateloader: 'a> base::Resolver for Resolver<'a, 'crateloader> {
                     if traits.is_empty() {
                         attrs.remove(i);
                     } else {
-                        let mut tokens = Vec::new();
+                        let mut tokens = Vec::with_capacity(traits.len().saturating_sub(1));
```

Comment:
No real need for the saturating_sub here since we've already checked if traits is empty (it's not)

Reply:
I'd say there's no real need to pollute code with reserves/capacities unless there are benchmarks.

Reply:
(Also, this specific code is being removed in https://github.com/rust-lang/rust/pull/54271.)

Reply:
@petrochenkov push loops with specified capacity are [_always_](https://gist.github.com/ljedrz/d85e3c4eb1c26b0ccfd100a1712d4cc9) faster than with an empty `Vec`.